akka-persistence-typed/src/main/scala/akka/persistence/typed/telemetry/EventSourcedBehaviorInstrumentation.scala

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.telemetry
+
+import java.util
+
+import akka.util.ccompat.JavaConverters._
+import scala.collection.immutable
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.event.Logging
+
+object EventSourcedBehaviorInstrumentation {
+  val EmptyContext: AnyRef = null
+}
+
+/**
+ * Instrumentation SPI for EventSourcedBehavior.
+ */
+trait EventSourcedBehaviorInstrumentation {
```

Comment:
Internal stable api sounds good.

I think either abstract class or (pure interface) trait. Don't think we should see any performance differences. For Cinnamon, abstract class was used, with the idea that performance could be better and fully optimised for monomorphic case. And the empty instrumentation was also a class, so it could be extended to only implement some of the SPI methods, but don't think we need to keep that. Using trait looks good to me for the these new SPIs.